CONTRIBUTING.md

```diff
@@ -47,44 +48,47 @@ the expected behavior that doesn't occur.
-`enhancement`.  Please do this before implementing a new feature to discuss the
-feature first.
+`enhancement`.  Please do this before implementing a new feature to discuss
+the feature first.
 
 
 ### Open a Pull Request
 
-Before submitting a PR, please read the NGINX Unit code guidelines to know more
-about coding conventions and benchmarks.  Fork the repo, create a branch, and
-submit a PR when your changes are tested and ready for review.  Again, if you'd
-like to implement a new feature, please consider creating a feature request
-issue first to start a discussion about the feature.
+Before submitting a PR, please read the NGINX Unit code guidelines to know
+more about coding conventions and benchmarks.  Fork the repo, create a branch,
+and submit a PR when your changes are tested and ready for review.  Again, if
+you'd like to implement a new feature, please consider creating a feature
+request issue first to start a discussion about the feature.
 
 
 ## Git Style Guide
 
-- Keep a clean, concise and meaningful `git commit` history on your branch,
-  rebasing locally and squashing before submitting a PR
+- Create atomic commits.  A commit should do just one thing, I.e you
+  shouldn't mix whitespace changes with actual code changes.  Do the
+  whitespace changes in a separate commit first.
 
-- For any user-visible changes, updates, and bugfixes, add a note to
-  `docs/changes.xml` under the section for the upcoming release, using `<change
-  type="feature">` for new functionality, `<change type="change">` for changed
-  behavior, and `<change type="bugfix">` for bug fixes.
+  Ideally you should rebase locally and force push new commits up.
 
-- In the subject line, use the past tense ("Added feature", not "Add feature");
-  also, use past tense to describe past scenarios, and present tense for
-  current behavior
+- In the subject line, use the imperative mood.  I.e write the subject like
+  you're giving git a command, e.g "Free memory before exiting". Do _not_
+  terminate the subject with a `.`
 
-- Limit the subject line to 67 characters, and the rest of the commit message
-  to 80 characters
+- Try to limit the subject line to around 50 characters, but try not to
+  exceed 72.
 
-- Use subject line prefixes for commits that affect a specific portion of the
-  code; examples include "Tests:", "Packages:", or "Docker:", and also
-  individual languages such as "Java:" or "Ruby:"
+- Wrap the body of the commit message after 72 characters.
 
-- Reference issues and PRs liberally after the subject line; if the commit
-  remedies a GitHub issue, [name
-  it](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
-  accordingly
+- Use lowercase subject line prefixes for commits that affect a specific
+  portion of the code; examples include "tests:", "ci:", or "http:", and
+  also individual languages such as "python:" or "php:".  Also if multiple
+  areas are affected, e.g "auto, perl:"
```

Comment:
```suggestion
  prefixes if multiple areas are affected, e.g. "auto, perl:"
```

Reply:
Hmm, that suggestion doesn't make sense to me...

Reply:
I've _tweaked_ it anyway...

Reply:
This suggestion goes hand in hand with the Combine one above this. Individually they won't make sense.